### PR TITLE
Added Homepage

### PR DESCRIPTION
**Pull request by vrinda-mahajan**
In this PR:-

- I have added the home page,
- basic routing,
- custom hook for document's title,
- category reducer + context

**Files to check**
src/App.js

**Components**
src/components/footer/footer.jsx
src/components/header/header.jsx
src/components/navbar/navbar.jsx
src/components/card/best-seller-card/best-seller-card.jsx
src/components/card/feature-card/feature-card.jsx

**Pages**
src/pages/homepage/homepage.jsx
src/pages/homepage/sub components/best-seller-section/best-seller-section.jsx
src/pages/homepage/sub components/featured-section/featured-section.jsx

**Context**
src/contexts/category-context.jsx

**Custom Hooks**
src/custom-hooks/useDocumentTitle.js

Also, review the folder structure.
Thank you

### CODE REVIEW

File: src/components/card/feature-card/feature-card.jsx [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import React from "react"
+
+export function FeatureCard ({img,category}) {
+    return(
+        <div >
+              <img className="featured-img" src={img} alt="Planters" />
+              <div className="featured-overlay">
+                  <h4>{category}</h4>
+                  <a href="./Pages/ProductPage/ProductPage.html"><button className="btn btn-primary btn-featured">Shop Now</button></a>
```
Comment (ApurvaSawant11) [inline src/components/card/feature-card/feature-card.jsx]: Use `Link` tag instead of `a` tag

File: src/components/navbar/navbar.jsx [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    return(
+        <nav className="nav-container align-center">
+  
+          <div className="p2 nav-logo flex-r">
+              <i className="app-icon fab fa-pagelines"></i>
+              <span className="app-name">FloraShop</span>
+          </div>
+  
+          <div className="nav-link">
+          <a href="#"><button className="btn btn-link nav-btn">Home</button></a>
```
Comment (ApurvaSawant11) [inline src/components/navbar/navbar.jsx]: Here as well, use `Link` tags

**Review by ApurvaSawant11 — APPROVED** [review]
What I liked about your code:
- Division of Home page into  different components
- Use of destructuring throughout the code

Suggestions:
- In the future, you can extract the reducer from the `category-context`. That way, code will look much cleaner and will be reusable

File: src/components/navbar/navbar.jsx (hunk 2 above)
Comment (vrinda-mahajan) [reply src/components/navbar/navbar.jsx]: Thanks for reviewing my PR. I have made the suggested changes.

**Review by MayankKumar10 — APPROVED** [review]
Just work on changes Suggested by Apurva, Everything else Looks Good to me you can merge it.

**Pull request merged**